modepy/quadrature/__init__.py

```diff
@@ -82,3 +102,79 @@ def __init__(self, quad, left, right):
         Quadrature.__init__(self,
                 left + (quad.nodes+1) / 2 * length,
                 quad.weights * half_length)
+
+
+class TensorProductQuadrature(Quadrature):
+    """
+    .. automethod:: __init__
+    """
+
+    def __init__(self, quads):
+        """
+        :arg quad: a :class:`tuple` of :class:`Quadrature` for one-dimensional
+            intervals, one for each dimension of the tensor product.
+        """
+
+        from modepy.nodes import tensor_product_nodes
+        x = tensor_product_nodes([quad.nodes for quad in quads])
+        w = np.prod(tensor_product_nodes([quad.weights for quad in quads]), axis=0)
+        assert w.size == x.shape[1]
+
+        super().__init__(x, w)
+        self.exact_to = min(quad.exact_to for quad in quads)
```

Comment:
This looks false. What's the combined order?

Reply:
To my mind, the real problem is that `exact_to` is not well-defined, at least not yet. I propose we define it as "exactly integrates up to *this* (summed) degree in multiple dimensions". (and record it in the docs) Under that definition, I think `min` is correct.

Reply:
How come you suspect it to be wrong?

Reply:
I just chose `min` out of pessimism and then realized that it may require some actual thought.

Added some docs in 099e587.

Reply:
Thanks! Commented there. Any other reason why this wouldn't live up to the definition I proposed?

Reply:
(Superseded by discussion below.)